icechunk/src/storage/mod.rs

```diff
@@ -222,47 +222,30 @@ where
     s.try_filter_map(|info| async move { Ok(convert_list_item(info)) }).boxed()
 }
 
-pub async fn make_storage(
-    config: ObjectStoreConfig,
-    bucket: Option<String>,
+pub async fn make_s3_storage(
+    config: S3Options,
```

Comment:
Should this be in the s3 mod? And same for the object store impls?

Reply:
I'll move it. I thought it would  be easier to search in documentation if we put them here, but I think I can change that by `pub use`

Reply:
I did this  but I didn't like it and revert it. Technicalities don't allow me to do something nice with `pub use`. These are such important methods, used as the first thing by any IC user, that I think they need to be very discoverable. I don't want to bury them inside many different classes the user would have to know about.

So I kept them here and made them public at the library level, so now you can say `icechunk::new_s3_storage`